Orange/data/table.py

```diff
@@ -1202,6 +1205,7 @@ def _filter_values_indicators(self, filter):
                 sel = np.vectorize(f)(col)
             elif isinstance(f, (data_filter.FilterContinuous,
                                 data_filter.FilterString)):
+                col = get_str_column(col, isinstance(f, data_filter.FilterString))
```

Comment:
Why calling a (trivial) function instead of doing it right here? It further obscures the logic of this code, which is already too contrived.